vars/buildPlugin.groovy

```diff
@@ -41,132 +41,140 @@ def call(Map params = [:]) {
+                                error 'There were test failures; halting early'
                             }
-                            checkstyle(cp)
-                        }
-                        if (failFast && currentBuild.result == 'UNSTABLE') {
-                            error 'There were test failures; halting early'
-                        }
-                        if (doArchiveArtifacts) {
-                            if (incrementals) {
-                                String changelist = readFile(changelistF)
-                                dir(m2repo) {
-                                    fingerprint '**/*-rc*.*/*-rc*.*' // includes any incrementals consumed
-                                    archiveArtifacts artifacts: "**/*$changelist/*$changelist*",
-                                            excludes: '**/*.lastUpdated',
-                                            allowEmptyArchive: true // in case we forgot to reincrementalify
-                                }
-                                publishingIncrementals = true
-                            } else {
-                                String artifacts
-                                if (isMaven) {
-                                    artifacts = '**/target/*.hpi,**/target/*.jpi'
+                            if (doArchiveArtifacts) {
+                                if (incrementals) {
+                                    String changelist = readFile(changelistF)
+                                    dir(m2repo) {
+                                        fingerprint '**/*-rc*.*/*-rc*.*' // includes any incrementals consumed
+                                        archiveArtifacts artifacts: "**/*$changelist/*$changelist*",
+                                                excludes: '**/*.lastUpdated',
+                                                allowEmptyArchive: true // in case we forgot to reincrementalify
+                                    }
+                                    publishingIncrementals = true
                                 } else {
-                                    artifacts = '**/build/libs/*.hpi,**/build/libs/*.jpi'
+                                    String artifacts
+                                    if (isMaven) {
+                                        artifacts = '**/target/*.hpi,**/target/*.jpi'
+                                    } else {
+                                        artifacts = '**/build/libs/*.hpi,**/build/libs/*.jpi'
+                                    }
+                                    archiveArtifacts artifacts: artifacts, fingerprint: true
                                 }
-                                archiveArtifacts artifacts: artifacts, fingerprint: true
                             }
                         }
+                    }            
+                } finally {
+                    deleteDir()
```

Comment:
Might be better to use https://jenkins.io/doc/pipeline/steps/ws-cleanup/

Reply:
Is that available on ci.jenkins.io? can change if you think it's worth it, I think it would be:
```suggestion
                    cleanWs
```

Reply:
Would be `cleanWs()`. Anyway I see no need for a special plugin, `deleteDir` works fine.

Do we actually need to do this? Just means that cloning will be slower (no update) if we do happen to reuse workspaces. I doubt that dirty workspaces are actually an issue (I have never seen any such problems), but if so I would rather recommend

```sh
git clean -f -f -d -x
```

Reply:
@olblak says the plugin is not installed, so it is a separate story

Reply:
By the way another good practice for CI (to run before `clean`):

```groovy
if (isUnix()) {
    sh 'git add . && git diff --exit-code HEAD'
}
```

which will fail if the build modified any versioned files, or created build products not in the ignore list.

Reply:
> Would be `cleanWs()`. Anyway I see no need for a special plugin, `deleteDir` works fine.
> 
> Do we actually need to do this? Just means that cloning will be slower (no update) if we do happen to reuse workspaces. I doubt that dirty workspaces are actually an issue (I have never seen any such problems), but if so I would rather recommend
> 
> ```shell
> git clean -f -f -d -x
> ```

Happy to integrate if required, but not sure if needed?